diff -urNp old/babl/babl-memory.c new/babl/babl-memory.c
--- old/babl/babl-memory.c	2017-10-06 09:07:51.085178917 +0200
+++ new/babl/babl-memory.c	2017-10-06 09:14:08.186524320 +0200
@@ -84,14 +84,19 @@ functions_sanity (void)
   if (first_malloc_used != malloc_f ||
       first_free_used != free_f)
     {
+      static int displayed = 0;
+
       if (first_malloc_used == NULL)
         {
           first_malloc_used = malloc_f;
           first_free_used   = free_f;
         }
-      else
+      else if (!displayed)
         {
-          babl_fatal ("babl memory function(s) attempted switched on the fly");
+          fprintf (stderr, "HMM....\nSomething strange is happening,\n%s function pointer changing between invocations in babl.\n",
+                  first_malloc_used == malloc_f ? "free" :
+                  (first_free_used == free_f ? "malloc" : "malloc and free"));
+          displayed = 1;
         }
     }
 }
